collect_app/src/main/java/org/odk/collect/android/external/ExternalDataUtil.java

```diff
@@ -205,6 +208,16 @@ public static XPathFuncExpr getSearchXPathExpression(String appearance) {
             }
             return returnedChoices;
         } catch (Exception e) {
+            String fileName = xpathfuncexpr.args[0].eval(null, null) + ".csv";
```

Comment:
At this point the filename hasn't been normalized by `ExternalDataHandlerBase.normalize` so it could include a `.csv` suffix, right? I think you need to check whether it's already there or not before adding it.

Reply:
Right, I thought in search you can use just a file name without .csv but looks like both versions work:
`search('fruits')` and `search('fruits.csv')`
Fixed.